### PR TITLE
ci(macos): Force dSYM creation on release builds

### PR DESCRIPTION
**Pull request by BYK**
Fixes the release branch builds like https://github.com/getsentry/relay/runs/2975323304. GitHub Actions runners upgrded Rust to 1.53, which comes with https://github.com/rust-lang/cargo/pull/9298, a breaking change for us since we expect to have the dSYM file for our builds.

#skip-changelog

### CODE REVIEW

**Review by untitaker — APPROVED** [review]
Approving if you need to get your release out right now, but if not I'd wait for a more substantial review

**Comment by BYK** [comment]
>Approving if you need to get your release out right now, but if not I'd wait for a more substantial review

Thanks @untitaker! Since I have the Sentry and Snuba releases out, I'd rather get this out now rather than to wait.

**Pull request merged**
